spec/features/users/show.html.erb_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'rails_helper'
+
+RSpec.describe 'User Show', type: :feature do
+  before :each do
+    @user = User.create(
+      name: 'Gaby',
+      photo: 'bulma.jpg',
+      bio: 'Systems Engineering Student'
+    )
+    @post1 = Post.create(author: @user, title: 'Post 1', text: 'Post One')
+    post2 = Post.create(author: @user, title: 'Post 2', text: 'Post Two')
+    post3 = Post.create(author: @user, title: 'Post 3', text: 'Post Three')
+    post4 = Post.create(author: @user, title: 'Post 4', text: 'Post Four')
+    @posts = [post2, post3, post4]
+    @posts_url = "a[href = '#{user_posts_path(@user.id)}']"
+
+    visit user_path(@user.id)
+  end
+
+  it 'shows the right content' do
+    expect(page).to have_content(@user.name)
+    expect(page).to have_css("img[src*='#{@user.photo}']")
+    expect(page).to have_content("Number of posts: #{@user.posts_counter}")
+    expect(page).to have_content(@user.bio)
+  end
+
+  it 'shows the last three posts' do
+    @posts.each do |post|
+      expect(page).to have_content(post.title)
+    end
+    expect(page).to_not have_content(@post1.title)
+  end
+
+  it "shows redirects to the post's show page on click of any of the posts" do
+    @posts.each do |post|
+      post_anchor = "a[href='#{user_post_path(@user.id, post.id)}']"
+      find(post_anchor).click
+      expect(page).to have_content('Post Detail')
+      expect(page).to have_content(post.text)
+    end
+  end
+
+  it 'has the see all posts button' do
+    button = find("#{@posts_url} button")
+    expect(button).to have_text('See all posts')
+  end
+
+  it 'redirects to posts index on clicking of the See all posts button' do
+    find(@posts_url).click
+    new_post_anchor = find("a[href='#{new_user_post_path(@user.id)}']")
+    expect(new_post_anchor).to have_content('New Post')
+  end
```

Comment:
- [ ] For some reason, when I run rspec to check whether the specs run or not, I am getting this error. Kindly look into why this is the case.

![image](https://user-images.githubusercontent.com/86318284/216712783-1c6b1e19-3e80-4542-a97c-922944e5dd4f.png)

Reply:
Hi @Nemwel-Boniface

Thank you for your kind review. We seem not to be getting these errors from our end.

![image](https://user-images.githubusercontent.com/20554583/216716229-c9864747-1444-4a6a-979a-e0002935a8a2.png)